experiment_helpers/check_rmsThresh.m

```diff
@@ -1,30 +1,119 @@
     ylim([0 0.1])
     xlim([0 tAxis(end)/data.params.sr])
-    hline(rmsThresh,'k',':');
+
+    yGood = [params.limits(1,1) params.limits(1,1) params.limits(1,2) params.limits(1,2)];
+    yWarn = [params.limits(2,1) params.limits(2,1) params.limits(2,2) params.limits(2,2)];
+
+    colorWarn = [1,   1,   0.3];
+    colorGood = [0.4, 1,   0.4];
+
+    xPatchShade = [0, length(data.rms), length(data.rms), 0];
+
+    patch(xPatchShade,yWarn,colorWarn, 'FaceAlpha', 0.3, 'EdgeColor', 'none')
+    patch(xPatchShade,yGood,colorGood, 'FaceAlpha', 0.3, 'EdgeColor', 'none')
+
+    %% If evaluating based on peak amplitude, do that
+    switch params.checkMethod
+        case 'peak'
+            rmsValue = max(data.rms(:,1));
+        case 'mean'
+            %% determine onset and offset
+            % if OST onset and offset exist, use that
+            if any(data.ost_stat == 4)
+                % Finding the last instance of status 0 implies that the next status
+                % was 1, and did eventually successfully become status 2.
+                onset = 1 + find(data.ost_stat == 0, 1, 'last');
+                offset = 1 + find(data.ost_stat == 2, 1, 'last');
+                rmsValue = mean(data.rms(onset:offset, 1));
+
+                % if no ost tracking, use RMS data to find onset/offset
+            elseif ~any(data.ost_stat >= 1) && any(data.rms(:, 1) > 0.03)
+                onset = find(data.rms > 0.01, 1, 'first') + 5;
+                offset = find(data.rms(:, 1)<0.03 & data.rms(:, 1)>0.02 & data.rms_slope<0, 1, 'first') - 5;
+
+                % use middle 80%
+                onset = floor(onset + ((offset-onset)/10));
+                offset = ceil(offset - ((offset-onset)/10));
+                rmsValue = mean(data.rms(onset:offset, 1));
+
+                % can't determine rms mean, because no ost-based vowel found, and RMS too low
+            else
+                rmsValue = NaN;
+            end
+
+            %% plot amplitude data
+            %plot mean RMS during vowel as a bar whose x position is vowel onset/offset, and y position is mean RMS
+
+            % TODO see if LineWidth is good. Default is 0.5
+            if ~isnan(rmsValue) && ~isempty(onset) && ~isempty(offset)
+                hold on;
+                xRMS = [onset offset] / length(data.rms) * max(tAxis) / data.params.sr;  %convert from frames to milliseconds
+                plot(xRMS, [rmsValue, rmsValue], '-', 'm')
+                hold off;
+            end
+    end
+
+    %% plot OST status
     yyaxis right
     plot(tAxis/data.params.sr,data.ost_stat);
-    if ~bGoodTrial
-        title({'';'Amplitude below threshold!'})
-    else
+    ylim([0 max(data.ost_stat + 2)]);
+
+    %% set title and bGoodTrial
+    if rmsValue > params.rmsThresh
+        bGoodTrial = 1;
         title({'',''})
-    end %of GoodTrial conditional
+    else
+        bGoodTrial = 0;
+        title({'';'Amplitude below threshold!'})
+    end
```

Comment:
Similar to above; I don't think bGoodTrial should be set inside this isgraphics if statement. We should set it first, then plotting is separate. It doesn't change functionality so I'm okay making this adjustment later.